include/rcpputils/filesystem_helper.hpp

```diff
@@ -536,6 +537,62 @@ inline bool remove(const path & p)
 #endif
 }
 
+/**
+ * \brief Remove the directory at the path p and its content.
+ *
+ * Additionally to \sa remove, remove_all removes a directory and its containing files.
+ *
+ * \param The path of the directory to remove.
+ * \return true if the directory exists and it was successfully removed, false otherwise.
+ */
+inline bool remove_all(const path & p)
+{
+  if (!is_directory(p)) {return false;}
+
+#ifdef _WIN32
+  // We need a string of type PCZZTSTR, which is a double null terminated char ptr
+  size_t length = p.string().size();
+  TCHAR * temp_dir = new TCHAR[length + 2];
```

Comment:
shouldn't be a `PCZZTSTR` instead of a `TCHAR *`?

Reply:
> PCZZTSTR may sound scary, but essentially it's just a simple C-style string of TCHARs, which is terminated with two '\0' characters. P (pointer to a), CZ (C-style, zero-terminated) Z (doubly zero-terminated), T (TCHAR), STR (string).

https://stackoverflow.com/a/11010402/4583130

Reply:
sure. But if `PCZZTSTR` is equivalent to `TCHAR *` double null terminated, why not using that type?
I don't mind much though.